layer/unittest/trace_event_log_tests.cc

```diff
@@ -27,8 +31,167 @@ using ::testing::IsEmpty;
 
 namespace performancelayers {
 namespace {
-// An `InstantEvent`. Adds a `scope` variable to the args indicating event's
-// visibility in the trace viewer.
+std::string ValueToJsonString(bool value) { return value ? "true" : "false"; }
+
+std::string ValueToJsonString(int64_t value) { return std::to_string(value); }
+
+std::string ValueToJsonString(const std::vector<int64_t> &values) {
+  std::stringstream json_str;
+  json_str << "[";
+  size_t e = values.size();
+  for (size_t i = 0; i != e; ++i) {
+    const char *delimiter = i < e - 1 ? "," : "";
+    json_str << "\"0x" << std::hex << values[i] << "\"" << delimiter;
```

Comment:
Wouldn't it be enough to pass `std::hex` only once (before the loop)?

Reply:
True. I think I should also add a `std::dec` at the end when I'm done writing. Right?

Reply:
I don't think you need to pass `dec` because the stringstream is discarded at the end

Reply:
I see.